src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/ValidateAnomalyDetectorTransportAction.java

```diff
@@ -0,0 +1,225 @@
+            resolveUserAndExecute(
+                user,
+                AnomalyDetector.NO_ID,
+                filterByEnabled,
+                listener,
+                () -> validateExecute(request, user, listener),
+                client,
+                clusterService,
+                xContentRegistry,
+                anomalyDetector,
+                false,
+                false
+            );
+        } catch (Exception e) {
+            logger.error(e);
+            listener.onFailure(e);
+        }
+    }
+
+    private void validateExecute(
+        ValidateAnomalyDetectorRequest request,
+        User user,
+        ActionListener<ValidateAnomalyDetectorResponse> listener
+    ) {
+        ActionListener<ValidateAnomalyDetectorResponse> validateListener = ActionListener.wrap(response -> {
+            logger.info("Result of validation process " + response);
+            // forcing response to be empty
+            listener.onResponse(new ValidateAnomalyDetectorResponse((DetectorValidationIssue) null));
+        }, exception -> {
+            if (exception instanceof ADValidationException) {
+                DetectorValidationIssue issue = parseADValidationException((ADValidationException) exception);
+                listener.onResponse(new ValidateAnomalyDetectorResponse(issue));
+                return;
+            }
+            listener.onFailure(exception);
+        });
+        ValidateAnomalyDetectorActionHandler handler = new ValidateAnomalyDetectorActionHandler(
+            clusterService,
+            client,
+            validateListener,
+            anomalyDetectionIndices,
+            request.getDetector(),
+            request.getRequestTimeout(),
+            request.getMaxSingleEntityAnomalyDetectors(),
+            request.getMaxMultiEntityAnomalyDetectors(),
+            request.getMaxAnomalyFeatures(),
+            RestRequest.Method.POST,
+            xContentRegistry,
+            user,
+            searchFeatureDao,
+            request.getTypeStr()
+        );
+        try {
+            handler.start();
+        } catch (Exception exception) {
+            String errorMessage = String
+                .format(Locale.ROOT, "Unknown exception caught while validating detector %s", request.getDetector());
+            logger.error(errorMessage, exception);
+            listener.onFailure(exception);
+        }
+    }
+
+    protected DetectorValidationIssue parseADValidationException(ADValidationException exception) {
+        String originalErrorMessage = exception.getMessage();
+        String errorMessage;
+        Map<String, String> subIssues = null;
+        Object suggestion = null;
+        switch (exception.getType()) {
+            case FEATURE_ATTRIBUTES:
+                // feature issues could be between square brackets like
+                // [Feature has issue: A, Feature has issue: B]
+                int firstLeftBracketIndex = originalErrorMessage.indexOf("[");
```

Comment:
Can we construct `DetectorValidationIssue` and put it into `ADValidationException` when validate ? So we can avoid complicated parsing logic here.

Reply:
> Can we construct `DetectorValidationIssue` and put it into `ADValidationException` when validate ? So we can avoid complicated parsing logic here.

I would prefer keeping constructing of `DetectorValidationIssue` out of `ADValidationException` during validating. `ADValidationException` is exception thrown during validation process, and `DetectorValidationIssue` is actually the issue we have identified, and return to user as response. They are separate concepts. Coupling them together might not be very necessary. I may prefer loose coupling as much as possible. 

Also, similar parsing logic is still needed even we move `DetectorValidationIssue` construction into process of validating. Consolidating parsing logic in single place is my personal preference. 

Please feel free to let me know if you still have concern.

Reply:
> Is it critical/necessary to split the error message into feature level? Is it enough to just show a general error message containing all feature level sub issues?

It is necessary from front-end side. If there are invalid multiple features, we need to identify which feature has issue, and show corresponding message to each feature definition.

Reply:
Got it. I'm ok with current logic. Let's check if we can simplify the logic later.